cpp/src/strings/strings_column_factories.cu

```diff
@@ -32,6 +32,134 @@
 
 namespace cudf {
 
+namespace strings::detail {
+std::vector<std::unique_ptr<column>> make_strings_column_batch(
+  std::vector<cudf::device_span<thrust::pair<char const*, size_type> const>> strings_batch,
+  rmm::cuda_stream_view stream,
+  rmm::device_async_resource_ref mr)
+{
+  std::vector<std::unique_ptr<column>> output;
+  std::vector<std::unique_ptr<column>>> offset_columns;
+  std::vector<size_type> total_bytes;
+  std::vector<size_type> strings_sizes;
+  std::vector<thrust::transform_iterator<size_type> offsets_transformer_itr;
+  std::vector<int64_t> chars_sizes;
+  std::vector<rmm::device_buffer> null_masks;
+  std::vector<size_type> null_counts;
+
+  std::transform(
+    strings_batch.begin(),
+    strings_batch.end(),
+    std::back_inserter(strings_sizes),
+    [] (auto &strings) {
+      return thrust::distance(strings.begin(), strings.end());
+    }
+  );
+
+  std::transform(
+    strings_batch.begin(),
+    strings_batch.end(),
+    std::back_inserter(offsets_transformer_itr),
+    [stream, mr] (auto &strings) {
+      size_type strings_count = thrust::distance(strings.begin(), strings.end());
+      auto offsets_transformer =
+        cuda::proclaim_return_type<size_type>([] __device__(string_index_pair item) -> size_type {
+          return (item.first != nullptr ? static_cast<size_type>(item.second) : size_type{0});
+        });
+      return thrust::make_transform_iterator(strings.begin(), offsets_transformer);
+    }
+  );
+
+  [offset_columns, total_bytes] = cudf::strings::detail::make_offsets_child_column_batch(
+    offsets_transformer_itr, strings_sizes, stream, mr);
```

Comment:
It seems impossible to store different types of `transform_iterator` into one vector because each element has a completely different type. So don't store and pass in a vector like that. Instead, pass in the input data (vectors of begin/end iterators).